server/src/main/java/com/paypal/selion/grid/MobileProcessLauncher.java

```diff
@@ -0,0 +1,121 @@
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+import static com.paypal.selion.pojos.SeLionGridConstants.*;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.annotations.Beta;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * A {@link RunnableLauncher} for mobile WebDrivers (appium, selendroid, and ios-driver). Adds the ability to process
+ * default arguments for the mobile WebDrivers (appium, selendroid, and ios-driver) from the SeLion Grid JSON config
+ * file.
+ */
+@Beta
+class MobileProcessLauncher extends AbstractBaseProcessLauncher {
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(MobileProcessLauncher.class);
+    private static final String SEPARATOR = "\n----------------------------------\n";
+    JsonObject defaultArgs;
+
+    public MobileProcessLauncher(String[] args) {
+        init(args);
+    }
+
+    public MobileProcessLauncher(String[] args, ProcessLauncherOptions options) {
+        init(args, options);
+    }
+    
+    /**
+     * Get program arguments to pass
+     *
+     * @return the program arguments to pass represented as an array of {@link String}
+     * @throws IOException
+     */
+    @Override
+    String[] getProgramArguments() throws IOException {
+        LOGGER.entering();
+        List<String> args = new LinkedList<String>();
+        args.addAll(Arrays.asList(super.getProgramArguments()));
+
+        // add the defaults which we don't already have a value for
+        Iterator<Entry<String, JsonElement>> iterator = defaultArgs.entrySet().iterator();
+        while (iterator.hasNext()) {
```

Comment:
Since we are interacting with every element in this Set, why not just use 
```
for (Map.Entry<String, JsonElement> eachArg : defaultArgs.entrySet()) {
...
}
```